client.go

```diff
@@ -125,17 +125,20 @@ func (c *Client) Send(req *http.Request, v interface{}) error {
 // making the main request
 // client.Token will be updated when changed
 func (c *Client) SendWithAuth(req *http.Request, v interface{}) error {
+	c.Lock()
 	if c.Token != nil {
 		if !c.tokenExpiresAt.IsZero() && c.tokenExpiresAt.Sub(time.Now()) < RequestNewTokenBeforeExpiresIn {
 			// c.Token will be updated in GetAccessToken call
 			if _, err := c.GetAccessToken(); err != nil {
+				c.Unlock()
```

Comment:
You are doing Unlock in multiple places, could look better if you do it as `defer c.Unlock()` right after Lock.

Reply:
I didn't want to include the `c.Send` in the critical section of that mutex. With a `defer c.Unlock()` we would essentially make all API requests sequential (which i don't think you would want?). 
Given this info, do you still want me to make that change?